lib/puppet/util/windows/adsi.rb

```diff
@@ -504,6 +504,39 @@ def self.current_user_name
       user_name
     end
 
+    # https://docs.microsoft.com/en-us/windows/win32/api/secext/ne-secext-extended_name_format
+    NameUnknown           = 0
+    NameFullyQualifiedDN  = 1
+    NameSamCompatible     = 2
+    NameDisplay           = 3
+    NameUniqueId          = 6
+    NameCanonical         = 7
+    NameUserPrincipal     = 8
+    NameCanonicalEx       = 9
+    NameServicePrincipal  = 10
+    NameDnsDomain         = 12
+    NameGivenName         = 13
+    NameSurname           = 14
+
+    def self.current_user_name_with_format(format)
```

Comment:
this method is always called with `Puppet::Util::Windows::ADSI::User::NameSamCompatible` as argument, I think it would be simpler and easier to read if we just use `NameSamCompatible` directly in this method when needed

Reply:
method might be renamed to current_sam_compatible_user_name

Reply:
That would've been my approach if we decided to replace the `current_user_name` method with this one. Then the `format` argument would've been defaulted to `NameSamCompatible`.

Due to the method's name (I would find it weird seeing `current_user_name_with_format` being called without arguments; the same way we already have `read_arbitrary_wide_string_up_to` with all 3 arguments defaulted), future proofing (other formats might be needed at some point and then we will either return to this implementation or have duplicated code) and to address your concern,  I would propose adding:
```ruby
def current_user_name_in_sam_format
  current_user_name_with_format(NameSamCompatible)
end
```